### PR TITLE
#415 clj deps support

### PR DESCRIPTION
**Pull request by mbuczko**
Support for dependecies described by deps.edn

### CODE REVIEW

**Comment by expez** [comment]
Hi, this looks good to me!

Could you squash this into one commit about clj deps support and add a `[Fix #415]` prefix to the new commit message?

We also need an integration test, to make sure the feature survives future changes to the code base, and a changelog entry.

Thanks for taking this on!

File: clj-refactor.el [hunk 1]
```diff
@@ -2073,19 +2082,22 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-project-clean"
...
-        cljr--get-sorted-dependency-names
-        (cljr--sort-dependency-vectors (thread-last (clojure-delete-and-extract-sexp)
-                                         (string-remove-prefix "[")
-                                         (string-remove-suffix "]")))
-        insert))
-    (indent-region (point-min) (point-max))
-    (save-buffer)))
+  (let ((project-file (cljr--project-file)))
+    (if (cljr--project-with-deps-p project-file)
+        (message "Dependencies sorting not supported in deps.edn yet.")
```
Comment (expez) [inline clj-refactor.el]: `user-error`

File: clj-refactor.el [hunk 2]
```diff
@@ -2181,18 +2193,34 @@ possible choices. If the choice is trivial, return it."
...
+      (goto-char (point-min))
+      (if depsp
+          (cljr--insert-into-project-deps artifact version)
+        (cljr--insert-into-project-dependencies artifact version))
+      (cljr--post-command-message "Added %s version %s as a project dependency" artifact version)
+      (when cljr-hotload-dependencies
+        (if depsp
+            (backward-sexp 2)
+          (paredit-backward-down))
+        (cljr-hotload-dependency)))))
```
Comment (expez) [inline clj-refactor.el]: This seems unlikely to work out of the box given the different data format.

Am I wrong?

File: clj-refactor.el (hunk 1 above)
Comment (mbuczko) [reply clj-refactor.el]: ok, done.

**Comment by expez** [comment]
Fantastic work! :+1: 

I suppose we should merge this at the same time as the changes to refactor-nrepl, to make sure hotloading works as it should?

File: clj-refactor.el [hunk 3]
```diff
@@ -2208,22 +2239,27 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-add-project-depe
...
 
 ;;;###autoload
-(defun cljr-update-project-dependency ()
+(defun cljr-update-project-dependency (&optional version)
   "Update the version of the dependency at point."
   (interactive)
   (cljr--ensure-op-supported "artifact-list")
-  (unless (cljr--looking-at-dependency-vector-p)
+  (unless (cljr--looking-at-dependency-p)
     (user-error "Place cursor in front of dependency vector to update."))
```
Comment (expez) [inline clj-refactor.el]: This error message might be slightly out of date now, as the edn based deps aren't vectors

File: clj-refactor.el [hunk 4]
```diff
@@ -2843,13 +2885,25 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-add-missing-libs
   (cljr--maybe-clean-ns)
   (cljr--maybe-eval-ns-form))
 
-(defun cljr--dependency-vector-at-point ()
+(defun cljr--dependency-at-point ()
+  "Returns project dependency at point.
+
+Recognizes both leiningen- and clj-style dependencies, but the later is always
```
Comment (expez) [inline clj-refactor.el]: `clj-style` -> deps.edn-style

later -> latter

File: clj-refactor.el (hunk 3 above)
Comment (mbuczko) [reply clj-refactor.el]: Would "Place cursor in front of dependency to update." be ok in this case?

**Comment by expez** [comment]
Great.

I found a few more nitpicks, but I think this is otherwise ready to get merged and for wider testing in the latest snapshot release.

I'll try to cut a snapshot release for refactor-nrepl later today and then we can merge this.

**Comment by benedekfazekas** [comment]
great work. let me know if you need help cutting the `refactor-nrepl` release @expez

File: clj-refactor.el (hunk 4 above)
Comment (mbuczko) [reply clj-refactor.el]: done.

**Comment by benedekfazekas** [comment]
nw. there is nothing more to do there right?

**Comment by expez** [comment]
>  is nothing more to do there right?

Right

**Comment by benedekfazekas** [comment]
took longer than expected, sorry. snapshot is on clojars.

**Comment by expez** [comment]
Merged! :fireworks: 

Thanks again, @mbuczko!

**Pull request merged**